tests/workflows/test_pytorch_optuna.py

```diff
@@ -26,9 +26,10 @@
             "torchvision",
         ],
         pip_options=[
-            "--force-reinstall",
             "--index-url",
             "https://download.pytorch.org/whl/cu118",
+            "--extra-index-url",
+            "https://pypi.org/simple",
```

Comment:
Without this, it was somehow making sqlalchemy unhappy with exactly this issue https://github.com/sqlalchemy/sqlalchemy/issues/10829. Prior to this PipInstall, the software environment would have the correct typing-extension>=4.6.0 (specifically 4.9.0) but this would manage to install a lower version.